### PR TITLE
Add some build stuff and fix packages

### PR DESCRIPTION
**Pull request by mikesplain**
Add dockerfile, update naming and make file.

This seems to have helped an issues I'm seeing with calico on k8s.  Not sure if you're interested in pushing to docker hub at some point, but that would be quite helpful!

### CODE REVIEW

**Comment by ottoyiu** [comment]
@mikesplain thanks for the PR! seems like glide.yaml needs to also be changed to reflect the correct github repo.

**Comment by ottoyiu** [comment]
seems like for whatever reason, in AWS, all the nodes just get purged :( have you seen that behaviour?

**Comment by mikesplain** [comment]
@ottoyiu Ahh yeah good catch. I had it right locally just hadn't pushed that yet :/.  I did see something funky but wasn't sure what caused it yet.  My goal was to first get it running in the cluster... then into fix mode :).  I'll look at this a bit today.  Let me know if you make any progress!

**Comment by mikesplain** [comment]
I think the issue is in the node names. I found after I fixed the cluster:
```
/ # calicoctl get nodes
NAME
ip-10-25-111-214
ip-10-25-32-40.ec2.internal
ip-10-25-53-140
ip-10-25-55-114.ec2.internal
ip-10-25-55-187
ip-10-25-67-254.ec2.internal
ip-10-25-69-251.ec2.internal
ip-10-25-70-207
ip-10-25-79-115
ip-10-25-98-153.ec2.internal
```

The nodes without `.ec2.internal` are current nodes registered by themselves.  In the logs for the controller I then see:
```
Node ip-10-25-111-214 does not exist anymore
Deleting node ip-10-25-111-214
```
for each of the real nodes currently in the cluster.  

TL;DR looks like theres an issue with the names of nodes getting compared without .ec2.internal.  I have to run out but will try to find the issue in a bit.

**Comment by ottoyiu** [comment]
That's what I'm seeing as well :( thanks for confirming my suspicion.

**Comment by mikesplain** [comment]
Not sure the best way to fix this... any ideas @ottoyiu? Looks like when calico starts up it uses the hostname of the node it's running on to register, which in our case is the non FQDN, without ec2.internal, yet nodes are registered in kubernetes with their full FQDN name.  I playing around with a fix for this but struggling to find the best spot...

**Comment by ottoyiu** [comment]
@mikesplain I'm planning to refactor the code and add a FQDN check and do comparisons on nodes.

Since Casey mentioned in the README that he does not intend to maintain this and continue development on this, I'm going to be working in this new repo:
https://github.com/ottoyiu/kube-calico-controller

Also, I'm not so sure what the value this would be once we switch over to Kubernetes API as the calico datastore.
